app/components/admin/editions/show/sidebar_actions_component.rb

```diff
@@ -0,0 +1,177 @@
+        href: confirm_force_schedule_admin_edition_path(@edition, lock_version: @edition.lock_version),
+        secondary_quiet: true,
+      })
+    end
+  end
+
+  def add_publish_action
+    if @publisher.can_perform? && @enforcer.can?(:publish)
+      actions << form_with(url: publish_admin_edition_path(@edition, lock_version: @edition.lock_version), method: :post) do
+        render("govuk_publishing_components/components/button", {
+          text: "Publish",
+          title: "Publish #{@edition.title}",
+        })
+      end
+    elsif @force_publisher.can_perform? && @enforcer.can?(:force_publish)
+      actions << render("govuk_publishing_components/components/button", {
+        text: "Force publish",
+        title: "Publish #{@edition.title}",
+        href: confirm_force_publish_admin_edition_path(@edition, lock_version: @edition.lock_version),
+        secondary_quiet: true,
+      })
+    end
+  end
+
+  def add_reject_action
+    if @edition.can_reject? && @enforcer.can?(:reject)
+      actions << form_with(url: reject_admin_edition_path(@edition, lock_version: @edition.lock_version), method: :post) do
+        render("govuk_publishing_components/components/button", {
+          text: "Reject",
+          destructive: true,
+        })
+      end
+    end
+  end
+
+  def add_destroy_action
+    if @edition.can_delete? && @edition.unpublishing.nil?
+      actions << link_to(
+        "Discard draft",
+        confirm_destroy_admin_edition_path(@edition),
+        class: "govuk-link gem-link--destructive",
+      )
+    end
+  end
+
+  def add_unwithdraw_action
+    if @enforcer.can?(:unwithdraw) && @edition.can_unwithdraw?
+      actions << render("govuk_publishing_components/components/button", {
+        text: "Unwithdraw",
+        href: confirm_unwithdraw_admin_edition_path(@edition, lock_version: @edition.lock_version),
+        secondary_quiet: true,
+      })
+    end
+  end
+
+  def add_unpublish_action
+    if @enforcer.can?(:unpublish)
+      if @edition.published?
+        actions << render("govuk_publishing_components/components/button", {
+          text: "Withdraw or unpublish",
+          href: confirm_unpublish_admin_edition_path(@edition, lock_version: @edition.lock_version),
+          destructive: true,
+        })
+      elsif @edition.unpublishing.present? && @edition.unpublishing.explanation.present?
+        actions << link_to(
+          "Edit #{helpers.withdrawal_or_unpublishing(@edition)} explanation",
+          edit_admin_edition_unpublishing_path(@edition),
+          class: "govuk-link",
+        )
+      end
+    end
+  end
+
+  def add_data_action
+    if @edition.publicly_visible?
+      actions << link_to("View on website (opens in new tab)", @url_maker.public_document_url(@edition), class: "govuk-link", target: "_blank", rel: "noopener")
+      actions << link_to("View data about page", helpers.content_data_page_data_url(@edition), class: "govuk-link", data: {
```

Comment:
This is super super nitty so feel free to ignore, but i wonder where this should be two methods as it actually adds two actions. 
```
def add_view_action; end
def add_data_action; end
```

Or pluralise it 

```
def add_data_actions; end
```

It's a tiny thing though. I only noticed as i was playing with generating the objects returned by methods within the array rather than being piped in and had to pop them in an array and flatten it.

Reply:
Thanks, I've added another method